mir_eval/transcription.py

```diff
@@ -0,0 +1,409 @@
+                                                    np.log2(est_pitches)))
+    pitch_hit_matrix = cmp_func(pitch_distances, pitch_tolerance)
+
+    # check for offset matches if offset_ratio is not None
+    if offset_ratio is not None:
+        offset_distances = np.abs(np.subtract.outer(ref_intervals[:, 1],
+                                                    est_intervals[:, 1]))
+        # Round distances to a target precision to avoid the situation where
+        # if the distance is exactly 50ms (and strict=False) it erroneously
+        # doesn't match the notes because of precision issues.
+        offset_distances = np.around(offset_distances, decimals=N_DECIMALS)
+        ref_durations = util.intervals_to_durations(ref_intervals)
+        offset_tolerances = np.maximum(offset_ratio * ref_durations,
+                                       offset_min_tolerance)
+        offset_hit_matrix = \
+            cmp_func(offset_distances, offset_tolerances.reshape(-1, 1))
+    else:
+        offset_hit_matrix = True
+
+    # check for overall matches
+    note_hit_matrix = onset_hit_matrix * pitch_hit_matrix * offset_hit_matrix
+    hits = np.where(note_hit_matrix)
+
+    # Construct the graph input
+    # Flip graph so that 'matching' is a list of tuples where the first item
+    # in each tuple is the reference note index, and the second item is the
+    # estimate note index.
+    G = {}
+    for ref_i, est_i in zip(*hits):
+        if est_i not in G:
+            G[est_i] = []
+        G[est_i].append(ref_i)
+
+    # Compute the maximum matching
+    matching = sorted(util._bipartite_match(G).items())
+
+    return matching
+
+
+def precision_recall_f1(ref_intervals, ref_pitches, est_intervals, est_pitches,
+                        onset_tolerance=0.05, pitch_tolerance=50.0,
+                        offset_ratio=0.2, offset_min_tolerance=0.05,
+                        strict=False):
+    """Compute the Precision, Recall and F-measure of correct vs incorrectly
+    transcribed notes. "Correctness" is determined based on note onset, pitch
+    and (optionally) offset: an estimated note is assumed correct if its onset
+    is within +-50ms of a ref note and its pitch (F0) is within +- quarter tone
+    (50 cents) of the corresponding reference note. If with_offset is False,
+    note offsets are ignored in the comparison. It with_offset is True,
+    on top of the above requirements, a correct returned note is required to
+    have an offset value within 20% (by default, adjustable via the
+    offset_ratio parameter) of the ref note's duration around the ref note's
+    offset, or within offset_min_tolerance (50 ms by default), whichever is
+    larger.
+
+    Examples
+    --------
+    >>> ref_intervals, ref_pitches = mir_eval.io.load_valued_intervals(
+    ...     'reference.txt')
+    >>> est_intervals, est_pitches = mir_eval.io.load_valued_intervals(
+    ...     'estimated.txt')
+    >>> precision, recall, f_measure =
```

Comment:
This would actually result in a syntax error.  Sadly, you need to do
```Python
>>> (precision,
...  recall,
...  f_measure) = mir_eval.transcription.precision_recall_f1(
...      ref_interals, ref_pitches, est_intervals, est_pitches)
```
You could also just do
```Python
>>> p, r, f = mir_eval.transcription.precision_recall_f1(
...      ref_intervals, ref_pitches, est_intervals, est_pitches)
```